src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1377,6 +1376,22 @@ private fun checkConstGenerics(holder: RsAnnotationHolder, constParameter: RsCon
     feature.check(holder, constParameter, "const generics")
 }
 
+private fun checkConstGenericsDefaults(holder: RsAnnotationHolder, default: RsExpr?) {
+    if (default == null) return
+    CONST_GENERICS_DEFAULTS.check(holder, default, "const generics defaults")
+}
+
+private fun checkConstArguments(holder: RsAnnotationHolder, args: List<RsExpr>) {
+    if (args.none { it is RsBinaryExpr }) return
+    val availability = CONST_GENERICS_DEFAULTS.availability(args.first())
+    if (availability != AVAILABLE) return
+    for (expr in args) {
+        if (expr is RsBinaryExpr) {
```

Comment:
Not only binary expressions should be annotated in this way
Try the following code:
```rust
#![feature(adt_const_params)]
#![feature(const_generics_defaults)]

#[derive(Eq, PartialEq)]
struct S0;
#[derive(Eq, PartialEq)]
struct S1(i32);
#[derive(Eq, PartialEq)]
struct S2 {
    x: i32
}

impl S2 {
    pub const fn new(x: i32) -> Self {
        S2 { x }
    }
}

const fn s1(x: i32) -> S1 {
    S1(x)
}

const C1: S1 = S1(123);

struct F0<const N: S0 = S0>();
struct F1<const N: S1 = S1(123)>();
struct F2<const N: i32 = C1.0>();
struct F3<const N: S1 = s1(123)>();
struct F4<const N: S2 = S2 { x: 123 } >();
struct F5<const N: S2 = S2::new(123)>();
```

Reply:
Ouch, sorry, I was about default params, not about args. But anyway, looks like not only binary expressions should be annotated here

Reply:
The compiler cannot parse and annotate some expressions in the const argument context. Our plugin does the same.